qbft/commit.go

```diff
@@ -82,35 +88,31 @@ func CreateCommit(state *State, config IConfig, root [32]byte) (*SignedMessage,
 
 		Root: root,
 	}
-	sig, err := config.GetShareSigner().SignRoot(msg, types.QBFTSignatureType, state.Share.SharePubKey)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed signing commit msg")
-	}
-
-	signedMsg := &SignedMessage{
-		Signature: sig,
-		Signers:   []types.OperatorID{state.Share.OperatorID},
-		Message:   *msg,
-	}
-	return signedMsg, nil
+	return Sign(msg, state.CommitteeMember.OperatorID, config.GetOperatorSigner())
 }
 
 func baseCommitValidationIgnoreSignature(
-	signedCommit *SignedMessage,
+	signedCommit *types.SignedSSVMessage,
 	height Height,
 	operators []*types.Operator,
 ) error {
-	if signedCommit.Message.MsgType != CommitMsgType {
-		return errors.New("commit msg type is wrong")
-	}
-	if signedCommit.Message.Height != height {
-		return errors.New("wrong msg height")
-	}
 
 	if err := signedCommit.Validate(); err != nil {
 		return errors.Wrap(err, "signed commit invalid")
 	}
 
+	msg, err := DecodeMessage(signedCommit.SSVMessage.Data)
+	if err != nil {
+		return err
+	}
+
+	if msg.MsgType != CommitMsgType {
+		return errors.New("commit msg type is wrong")
+	}
+	if msg.Height != height {
```

Comment:
similar to previous check, shouldn't this be checked before calling the function?

Reply:
Not sure what you mean...
This is simply part of validation